packages/context/src/binding-key.ts

```diff
@@ -103,4 +103,17 @@ export class BindingKey<ValueType> {
       keyWithPath.substr(index + 1),
     );
   }
+
+  static CONFIG_NAMESPACE = '$config';
+  /**
+   * Build a binding key for the configuration of the given binding.
+   * The format is `<key>:$config`
+   *
+   * @param key The binding key that accepts the configuration
+   */
+  static buildKeyForConfig(key: BindingAddress<unknown> = '') {
```

Comment:
Can we modify this function to support type checks and inference?

```ts
export class BindingKey<ValueType> {
  // ...

  static buildKeyForConfig<T>(key: BindingAddress<unknown> = ''): BindingKey<T> {
    // ...
    return BindingKey.create<T>(bindingKey);
  }
}
```

Also since this method is part of `BindingKey`, can we perhaps shorten it's name to `forConfig`, with the following use?

```ts
const key = BindingKey.forConfig(...);
```

Reply:
> Can we modify this function to support type checks and inference?
> (...)
> Also since this method is part of BindingKey, can we perhaps shorten it's name to forConfig, with the following use?
> (...)

☝️